src/io.rs

```diff
@@ -2,85 +2,126 @@
 
 use core::{fmt, slice};
 use core::fmt::Write;
+use core::ptr;
 
 /// File descriptors
-const STDOUT: usize = 1;
-const STDERR: usize = 2;
+static mut STDOUT: isize = -1;
+static mut STDERR: isize = -1;
 
 /// Host's standard error
 struct Stderr;
 
 /// Host's standard output
 struct Stdout;
 
-fn write_all(fd: usize, mut buffer: &[u8]) {
+pub fn get_stdout() -> isize {
```

Comment:
IMO, this should be named `stdout` to match `std`'s naming convention and return a `Result` to indicate that you forgot to open the streams. While you are at it you may as well return `Result<Stdout>` where `Stdout` is non-instanceable (i.e. `struct Stdout { _private: () }`) zero type sized that has a `write_all` method (that has no `fd < 0` error branch) for writing to stdout.